datafusion/sql/src/planner.rs

```diff
@@ -2278,6 +2269,17 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         let mut result_days: i64 = 0;
         let mut result_millis: i64 = 0;
 
+        // Only handle string exprs for now
+        let value = match &value {
```

Comment:
I am not sure about this one -- now the arguments to `INTERVAL` can be an arbitrary expression rather than just a `String` (after https://github.com/sqlparser-rs/sqlparser-rs/pull/517)

cc @ovr any thoughts?